From 6cde84de6b7804ab858faf41bf4bad34899dce6e Mon Sep 17 00:00:00 2001
From: Devajith Valaparambil Sreeramaswamy
 <devajith.valaparambil.sreeramaswamy@cern.ch>
Date: Thu, 31 Oct 2024 09:32:17 +0100
Subject: [PATCH 12/52] Revert "Do not read the file to determine its name!"

This reverts commit e39e4edcac148b2625bc09d297524644ac56b1b1.
---
 .../llvm-project/clang/lib/Basic/SourceManager.cpp     | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/interpreter/llvm-project/clang/lib/Basic/SourceManager.cpp b/interpreter/llvm-project/clang/lib/Basic/SourceManager.cpp
index a5836bcdf1..c10d71c748 100644
--- a/interpreter/llvm-project/clang/lib/Basic/SourceManager.cpp
+++ b/interpreter/llvm-project/clang/lib/Basic/SourceManager.cpp
@@ -1462,16 +1462,6 @@ StringRef SourceManager::getBufferName(SourceLocation Loc,
   auto B = getBufferOrNone(getFileID(Loc));
   if (Invalid)
     *Invalid = !B;
-
-  // Try to get the name without reading the buffer.
-  FileID FID = getFileID(Loc);
-  const SrcMgr::SLocEntry &Entry = getSLocEntry(FID, Invalid);
-  if (!Invalid && Entry.isFile()) {
-    if (OptionalFileEntryRef FE =
-            Entry.getFile().getContentCache().ContentsEntry)
-      return FE->getName();
-  }
-
   return B ? B->getBufferIdentifier() : "<invalid buffer>";
 }
 
-- 
2.43.5

